### PR TITLE
fix(core): fix db crash after enabling dedup on varchar column

### PR DESCRIPTION
**Pull request by ideoma**

### CODE REVIEW

**Comment by ideoma** [comment]
> LGTM. Would be great to add a test vulnerable to the crash.

`DedupInsertFuzzTest.testRandomDedupRepeat` fails with the new assert sometimes.

**Comment by puzpuzpuz** [comment]
> `DedupInsertFuzzTest.testRandomDedupRepeat` fails with the new assert sometimes.

Thanks, I was able to reproduce the crash with the test.

**Review by puzpuzpuz — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 5 / 5 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/VarcharTypeDriver.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/StringTypeDriver.java|1|1|100.00%|

File: core/src/main/c/share/dedup.cpp [hunk 1]
```diff
@@ -667,6 +667,34 @@ Java_io_questdb_std_Vect_dedupMergeVarColumnLen(JNIEnv *env, jclass cl,
     return dst_var_offset;
 }
 
+
+JNIEXPORT jlong JNICALL
+Java_io_questdb_std_Vect_dedupMergeVarcharColumnLen(JNIEnv *env, jclass cl,
```
Comment (bluestreak01) [inline core/src/main/c/share/dedup.cpp:672]: this method should be called `dedupMergeVarcharColumnSize` for three reasons:
- everything that is called "size" in the driver provides size in bytes
- migrating to the driver i had very hard time establishing what is "size" and what is "row count", things were called "len" and "size" interchenageably and they could mean either row count or size depending on context

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

File: core/src/main/c/share/dedup.cpp [hunk 2]
```diff
@@ -667,6 +667,34 @@ Java_io_questdb_std_Vect_dedupMergeVarColumnLen(JNIEnv *env, jclass cl,
...
+                                                   jlong merge_index_addr,
+                                                   jlong merge_index_size,
```
Comment (bluestreak01) [inline core/src/main/c/share/dedup.cpp:674]: this parameter should be consistently called "merge_index_row_count", its value is literally derived from

```
timestampMergeIndexSize / TIMESTAMP_MERGE_ENTRY_BYTES
```

we cannot carry on calling this size after this division

File: core/src/main/c/share/dedup.cpp (hunk 1 above)
Comment (ideoma) [reply core/src/main/c/share/dedup.cpp]: Ok, done

File: core/src/main/c/share/dedup.cpp (hunk 2 above)
Comment (ideoma) [reply core/src/main/c/share/dedup.cpp]: Ok, done

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 5 / 5 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/VarcharTypeDriver.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/StringTypeDriver.java|1|1|100.00%|

**Pull request merged**